src/pages/MainPage/components/ProceedContainer/components/SumaryContainer/components/SumaryBody/components/TableBodyContainer/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import {
+  selectBasket,
+  setSubtotal,
+  setGrandTotal,
+  setShipping,
+} from "features/basket/basketSlice";
+import { Button } from "components";
+import { TableItem } from "./components";
+import * as Styles from "./styles";
+
+const TableBodyContainer = () => {
+  const dispatch = useDispatch();
+  const basket = useSelector(selectBasket);
+
+  useEffect(() => {
+    dispatch(setSubtotal(subtotal()));
+    dispatch(setShipping(shipping));
+    dispatch(setGrandTotal(subtotal() === 0 ? 0 : subtotal() + shipping));
+  }, [basket]);
+
+  const subtotal = () => {
+    let subtotal = 0;
+    [...basket.items].forEach((item) => {
+      const itemValue = item.price * item.value;
+      subtotal = subtotal + itemValue;
+    });
+
+    return subtotal;
+  };
+
+  const shipping = subtotal() <= 100 ? 23.8 : 0;
+
+  return (
+    <Styles.Wrapper>
+      <TableItem text={"Subtotal"} value={`$${basket.subtotal.toFixed(2)}`} />
```

Comment:
no i tu masz basket.subtotal, to czemu są dwa różne subtotal liczone w różnych miejscach :D

Reply:
Done.